### PR TITLE
Remove Airy init and restructure cli

### LINKED ISSUE #1051: Remove airy init
As `airy create` will effectively do all the work `airy init` does, the command is redundant from a UX perspective. 

We use this issue as an "excuse" to also restructure the CLI looks for config. Now we have different flags for different things (actual cli config, kube.conf.... airy.yaml file). I suggest we let users only choose a `config-dir` which will contain all the configuration of an airy core instance. 

That way we can also add some nice docs that suggest how to deal with multiple instances as you'd only need to point your cli to different directories and "everything just works" then (proposing some aliases in the docs would make this very nice to work with)

In the context of this change, we can also restructure the integration tests of the CLI a little:

- the runner generates a temp dir
- we put "golden files" in the temp dir
- we point the cli to this temp dir
- run all the tests in a single integration file 
- delete the temp dir

After the conversation in the comments, here's what we plan to do:

- [ ] introduce the concept of `configdir` while removing the file config flags (we point the CLI to a dir with `airy.yaml`, `kube.conf` `cli.yaml`)
- [ ] Because of the previous point, we refactor the integration tests as described (look up in this description)
- [x] move cli to the root directory (#1401)
- [ ] re-organise the packages internally (this point can be further discussed at pr time I think)
- [ ] remove the init command
- [ ] add the --init flag to the create command (so we cover the case @ljupcovangelski brought up)
- [ ] Update all the docs :)

### PR DESCRIPTION
**Pull request by chrismatix**
resolves #1051 

Check out the issue for a detailed list of changes.

### CODE REVIEW

File: cli/integration/BUILD [hunk 1]
```diff
@@ -8,6 +8,13 @@ go_library(
         "runner.go",
         "test_file.go",
     ],
+    embedsrcs = [
```
Comment (chrismatix) [inline cli/integration/BUILD:11]: > we put "golden files" in the temp dir
We can discuss how best to handle the golden files. I did not understand what would be the benefit of adding them to the temp config dir. Embedding them makes handling the paths a bit easier. 

@lucapette Could you help me understand the reasoning behind the quoted point a bit more? :)

Comment (lucapette) [reply cli/integration/BUILD:11]: because right now it changes the git status

File: cli/pkg/providers/aws/aws.go [hunk 2]
```diff
@@ -20,23 +21,29 @@ import (
...
 }
 
-func (a *Aws) GetHelmOverrides() []string {
+func New(w io.Writer) *provider {
+	return &provider{
+		w: w,
+	}
+}
+
+func (a *provider) GetHelmOverrides() []string {
```
Comment (lucapette) [inline cli/pkg/providers/aws/aws.go:42]: then you want to call it `p` :)

File: cli/pkg/providers/minikube/minikube.go [hunk 3]
```diff
@@ -18,20 +19,27 @@ const (
...
 }
 
-func (m *Minikube) GetHelmOverrides() []string {
+func New(w io.Writer) *provider {
+	return &provider{
+		w: w,
+	}
+}
+
+func (m *provider) GetHelmOverrides() []string {
```
Comment (lucapette) [inline cli/pkg/providers/minikube/minikube.go:33]: then you want to call this one `p`

File: cli/pkg/workspace/init.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	}
+
+	for _, entry := range entries {
+		if err := recCopy(path, "template", entry); err != nil {
+			return ConfigDir{}, err
+		}
+	}
+
+	// Init viper config
+	err = viper.WriteConfigAs(filepath.Join(path, cliConfigFileName))
```
Comment (lucapette) [inline cli/pkg/workspace/init.go:74]:
```
	if err != nil {
		return ConfigDir{}, err
	}
```

and then 

```
return ConfigDir{Path: path}, nil
```

would be more precise no? (as in you deal with the error right away)

File: docs/docs/cli/installation.md [hunk 5]
```diff
@@ -129,15 +129,6 @@ Make sure the output matches the version number you expect:
...
 
-### Step 4: Initialize the setup
-
-The [airy init](cli/reference.md#init) will create a `cli.yaml` file which stores
-your `apiHost` and `apiJwtToken`:
-
-```bash
-airy init
-```
-
```
Comment (lucapette) [inline docs/docs/cli/installation.md:140]: 😍

Comment (lucapette) [reply docs/docs/cli/installation.md:140]: but there's some sort of TOC in this file so we must remove more

**Review by lucapette — COMMENTED** [review]
Looking great!

File: cli/pkg/workspace/init.go (hunk 4 above)
Comment (chrismatix) [reply cli/pkg/workspace/init.go:74]: I will move the three commands in between above this one so the immediate return makes more sense :)

**Review by lucapette — APPROVED** [review]
:shipit:

**Pull request merged**